src/IW/Core/WithId.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE DeriveAnyClass #-}
+
+{- | @WithId a@ wraps values of type @a@ that don't have an @Id a@ field, but need one in
+order to satisfy roundtrip property tests. We use it to avoid adding an extra field of 
```

Comment:
I propose to rephrase the part about why it needs. Because the primary goal of `Withid` is to not to implement property tests but to be able to provide `Id` for a data types that don't know about their `Id`. Data types don't know about `Id` not because we wanted to implement `toRow` instance without outputting `Id` but because we fetch data from external sources and those sources don't have information about our `Id`. But later down the road, we might want to expose those `id` to the frontend, so we can do some things more efficiently, like _force repo to sync by Id_ or _update issue details by Id_.

Reply:
Just updated the docs. It's more concise. Let me know if it needs more information.